Assets/MixedRealityToolkit.SDK/Features/Utilities/Solvers/TapToPlace.cs

```diff
@@ -0,0 +1,365 @@
+        [Tooltip("This event is triggered once when the game object to place is selected.")]
+        private UnityEvent onPlacingStarted = new UnityEvent();
+
+        /// <summary>
+        /// This event is triggered once when the game object to place is selected.
+        /// </summary>
+        public UnityEvent OnPlacingStarted
+        {
+            get => onPlacingStarted;
+            set => onPlacingStarted = value;
+        }
+
+        [SerializeField, FormerlySerializedAs("OnPlacingStopped")]
+        [Tooltip("This event is triggered once when the game object to place is unselected, placed.")]
+        private UnityEvent onPlacingStopped = new UnityEvent();
+
+        /// <summary>
+        /// This event is triggered once when the game object to place is unselected, placed.
+        /// </summary>
+        public UnityEvent OnPlacingStopped
+        {
+            get => onPlacingStopped;
+            set => onPlacingStopped = value;
+        }
+
+        protected internal int GameObjectLayer { get; protected set; }
+
+        protected internal bool IsColliderPresent => gameObject != null ? gameObject.GetComponent<Collider>() != null : false;
+
+        private int ignoreRaycastLayer;
+
+        // The current ray is based on the TrackedTargetType (Controller Ray, Head, Hand Joint)
+        protected RayStep CurrentRay;
+
+        protected bool DidHitSurface;
+
+        protected RaycastHit CurrentHit;
+
+        protected float LastTimeClicked = 0;
+
+        protected float doubleClickTimeout = 0.5f;
+
+        #region MonoBehaviour Implementation
+        protected override void Start()
+        {
+            // Solver base class
+            base.Start();
+
+            Debug.Assert(IsColliderPresent, "The game object " + gameObject.name + " does not have a collider attached, please attach a collider to use Tap to Place");
+            
+            SurfaceNormalOffset = gameObject.GetComponent<Collider>().bounds.extents.z;
+
+            ignoreRaycastLayer = LayerMask.NameToLayer("Ignore Raycast");
+
+            if (AutoStart)
+            {
+                StartPlacement();  
+            }
+            else
+            {
+                SolverHandler.UpdateSolvers = false;
+            } 
+        }
+
+        private void OnDisable()
+        {
+            CoreServices.InputSystem?.UnregisterHandler<IMixedRealityPointerHandler>(this);
+        }
+
+        #endregion
+
+        /// <summary>
+        /// Start the placement of a game object without the need of the OnPointerClicked event. The game object will begin to follow the 
+        /// TrackedTargetType (Head by default) at a default distance. StopPlacement() must be called after StartPlacement() to stop the 
+        /// game object from following the TrackedTargetType.
+        /// </summary>
+        public void StartPlacement()
+        {
+            // Store the initial game object layer
+            GameObjectLayer = gameObject.layer;
```

Comment:
What happens if people call StartPlacement() twice in a row, or StopPlacement() twice in a row?

Reply:
Ah calling StartPlacement twice in a row did cause an issue. An IsBeingPlaced check has been added to Start/StopPlacement to make sure nothing happens if there are multiple calls.